archeryutils/handicaps/handicap_scheme.py

```diff
@@ -715,4 +730,5 @@ def _f_root(
 
         """
         val = self.score_for_round(hc_est, round_est, arw_d=arw_d, rounded_score=False)
-        return val - score_est
+        # val is known to be a 0D array, so cast to float for type checker
+        return cast(float, val) - score_est
```

Comment:
If `val` is a 0D array, `val - score_est` will be of type `np.float64`. Rather than using `cast`, if you instead do `float(val) - score_est`, the result will actually be of type `float`.

Reply:
Good call! thanks.